p2p/libp2p/netMessenger.go

```diff
@@ -645,6 +660,15 @@ func (netMes *networkMessenger) SetPeerShardResolver(
 	}
 	netMes.peerShardResolver = peerShardResolver
 
+	peerDiscovererWithSharder, ok := netMes.peerDiscoverer.(PeerDiscovererWithSharder)
+	if ok {
```

Comment:
If !ok return nil. Rest of the code after

Reply:
done